### PR TITLE
support task dag with gang

### PR DESCRIPTION
**Pull request by HecarimV**
[Task dependency in job](https://github.com/volcano-sh/volcano/blob/master/docs/design/task-launch-order-within-job.md) is supported in Volcano. 
However, the task depends function must remove the gang plugin to run normally.
I'm trying to get the task dag and gang plugin to work together. So I made some simple modifications.
- Special handling for Jobs with task dependencies when check  job valid. 
- keep pg inqueue when task dag is not finished.

Just a simple try, I want to discuss if it is possible.

Signed-off-by: HaoJie Liu <1187851227@qq.com>

### CODE REVIEW

**Comment by hwdef** [comment]
Please make CI happy first.

**Comment by HecarimV** [comment]
> Please make CI happy first.

okay, I'll check it.

File: pkg/controllers/job/job_controller_actions.go [hunk 1]
```diff
@@ -44,6 +44,10 @@ import (
 
 var calMutex sync.Mutex
 
+const (
+	customLaunch = "volcano.sh/custom-launch"
```
Comment (Thor-wl) [inline pkg/controllers/job/job_controller_actions.go:48]: BTW, what's the meaning of this annotation? Pls add some comments about that.

Comment (HecarimV) [reply pkg/controllers/job/job_controller_actions.go:48]: I want to use this annotation to mark jobs that have task dependencies，or just change the key as “has-task-dependencies”

**Comment by HecarimV** [comment]
> @HecarimV Thanks for your pr. It's a feature enhancement. Would you like to provide a doc based on "task dependency in job" firstly?

Okay, I will organize the ideas into documents.

**Pull request closed without merging**